### PR TITLE
Issue10294 new code for two column layout on typst 0.12.x 

### PR DESCRIPTION
**Pull request by jarnowic**
Typst 0.12.0 introduces some changes for two column layout, preferring `#page(columns: <N>)` for global two column output, and supplementing `#place` with new parameters `scope` and `clearance` to layout one column content (such as titleplage) on two column documents. This patch tries to upgrade the `typst.template` to comply with the newer requirements. Check [issue 10294](https://github.com/jgm/pandoc/issues/10294).

### CODE REVIEW

**Comment by jgm** [comment]
template.typst has a `set page` with other attributes. Why not put the `columns` there rather than in default.typst?

**Comment by jarnowic** [comment]
> template.typst has a `set page` with other attributes. Why not put the `columns` there rather than in default.typst?

because `default.typst` is the piece of code where the variables are accessed to from the YAML headers or the command line, iiuc. 

to tell the truth, i am still not quite sure why so many defaults are set in `template.typst`; it seems to me that only the variables to be used in customizations in the typst documents should be declared in the typst template, while the document layout decisions might be delayed to the document itself. looks to me as if `template.typst` should work not much unlike `latex.fmt`, just declaring the variables available, while all layout defaults should be moved into `default.typst` to give room for customization, not much unlike LaTeX classes and packages. my apologies for the comparison, but that's the model i am more familiar with; `groff_ms` is still, not much unlike `typst`, work in progress on my side.

so, if i were to suggest something, i'd say that the layout code should be moved into `default.typst` from `template.typst` to allow for more customizations, and not the other way around.

**Comment by jgm** [comment]
> template.typst is out of reach, so pandoc variables cannot affect it.

Not so.  You can put variables in partials!

**Comment by jarnowic** [comment]
> > template.typst is out of reach, so pandoc variables cannot affect it.
> 
> Not so. You can put variables in partials!

you're right; and doing so keeps the change less intrusive than the other one. proceeding.

**Comment by jarnowic** [comment]
> > > template.typst is out of reach, so pandoc variables cannot affect it.
> > 
> > 
> > Not so. You can put variables in partials!
> 
> you're right; and doing so keeps the change less intrusive than the other one. proceed

> > > template.typst is out of reach, so pandoc variables cannot affect it.
> > 
> > 
> > Not so. You can put variables in partials!
> 
> you're right; and doing so keeps the change less intrusive than the other one. proceeding.

I think it is ready now...

File: data/templates/default.typst [hunk 1]
```diff
@@ -31,12 +31,12 @@ $endif$
 $if(smart)$
 $else$
 #set smartquote(enabled: false)
-
 $endif$
+
```
Comment (jgm) [inline data/templates/default.typst:35]: Why did you move the endif here?  Putting the space before the endif is a feature: it means that we'll only get the additional blank line if we get the `#set smartquote` line.  Putting it after will give us a blank line even if there is nothing.

File: data/templates/default.typst [hunk 2]
```diff
@@ -31,12 +31,12 @@ $endif$
...
 $for(header-includes)$
 $header-includes$
-
 $endfor$
+
```
Comment (jgm) [inline data/templates/default.typst:39]: Ditto for endfor.

File: data/templates/template.typst [hunk 3]
```diff
@@ -35,55 +35,50 @@
     paper: paper,
     margin: margin,
     numbering: "1",
-  )
+    columns: cols,
+    )
+
```
Comment (jgm) [inline data/templates/template.typst:40]: why the blank line?  we don't have one elsewhere, e.g. btw 33 and 34

File: data/templates/template.typst [hunk 4]
```diff
@@ -35,55 +35,50 @@
...
            size: fontsize)
   set heading(numbering: sectionnumbering)
 
-  if title != none {
-    align(center)[#block(inset: 2em)[
-      #text(weight: "bold", size: 1.5em)[#title]
-      #(if subtitle != none {
+  place(top + center, float: true, scope: "parent", clearance: 4mm)[
+    #if title != none {
+      text(weight: "bold", size: 1.5em)[#title]
```
Comment (jgm) [inline data/templates/template.typst:50]: Can you say something about the rationale for this change in the way the title is formatted?

File: data/templates/template.typst [hunk 5]
```diff
@@ -35,55 +35,50 @@
...
 
-  if cols == 1 {
-    doc
-  } else {
-    columns(cols, doc)
-  }
+    #if abstract != none {
+      text(weight: "semibold")[Abstract]
+      h(1em) 
+      abstract
```
Comment (jgm) [inline data/templates/template.typst:80]: Can you confirm that this gives the same visual appearance as the previous template?  Before we had a block with 2em inset.  The `#block` has been removed.  What effect does that have?

File: data/templates/default.typst (hunk 1 above)
Comment (jarnowic) [reply data/templates/default.typst]: fair enough: reverted

File: data/templates/default.typst (hunk 2 above)
Comment (jarnowic) [reply data/templates/default.typst]: reverted, as for smartquote

File: data/templates/template.typst (hunk 3 above)
Comment (jarnowic) [reply data/templates/template.typst]: it was somewhat of an aid for me to debug the code; didn't seem to do any harm, but now I see the underlying rationales.

Comment (jarnowic) [reply data/templates/template.typst]: it was there, in the previous version, between the page settings and the titlepage code.

Comment (jarnowic) [reply data/templates/template.typst:40]: actually, it was there in the first place, it was just somewhat hidden from view in the change file...

File: data/templates/template.typst (hunk 4 above)
Comment (jarnowic) [reply data/templates/template.typst:50]: sure: in versions <0.12 column placement was somewhat assumed to be global on layout; with typst >=0.12 it is explicitly controlled by page settings, with an explicit "scope", so the previous code placed only the main text in two columns, while the footnotes were still single column. Changing the page layout to double column had the side effect to set the title page also in double column (it was moved to the first, left column in my test). This new code is the current standard way to set a block of text single spaced, outside the scope of the double column page layout.

File: data/templates/template.typst (hunk 5 above)
Comment (jarnowic) [reply data/templates/template.typst]: I believe it's all cleaned up now...

**Comment by jgm** [comment]
looks good - I'll merge this and update the tests.
